src/ol_infrastructure/components/aws/auto_scale_group.py

```diff
@@ -83,6 +83,7 @@ class OLTargetGroupConfig(AWSBase):
     health_check_protocol: str = "HTTPS"
     health_check_timeout: PositiveInt = PositiveInt(5)
     health_check_unhealthy_threshold: PositiveInt = PositiveInt(3)
+    max_instance_lifetime: Optional[PositiveInt] = 2592000  # 30 days
```

Comment:
This should include a unit in the variable name to avoid confusion. Also, this attribute will need to be passed to the actual ASG resource in the body of the component resource to have an effect.